docs/AddressSpace.md

```diff
@@ -0,0 +1,94 @@
+   These methods may enlarge the allocation further at the behest of the platform.
+
+7. If the global `AddressSpaceManager` enlarged the request, it will split the platform-provided memory region, caching the remainder (in its `AddressSpaceManagerCore`) for subsequent allocations.
+   (The free lists holding any remainder regions here are encoded into the Pagemap.)
+
+8. `Backend::reserve` then, again, may split off a piece of the provided region, storing the rest in its local `AddressSpaceManagerCore`.
+   (The free lists holding any remainder regions here are encoded into the Pagemap, again.)
+
+9. `Backend::alloc_chunk` additionally allocates a `Metaslab` and registers this with the `Pagemap`, together with the slab sizeclass value and a pointer to this `LocalAlloc`'s `RemoteAllocator` public state.
+
+10. `CoreAllocator::small_alloc_slow` constructs a free list in the slab and returns the whole list to the local `LocalCache`.
+
+11. `LocalAlloc::small_alloc` now takes the head of this free list and returns that allocation to the user.
+
+## What Can I Learn from the Pagemap?
+
+The centerpiece of `snmalloc`'s metadata is its `PageMap`, which associates each "chunk" of the address space (~16KiB; see `MIN_CHUNK_BITS`) with a `MetaEntry`.
+A `MetaEntry` is a pair of pointers; the first uses offset manipulation to densely encode...
+
+1. A `sizeclass` field encoding one of...
+
+   1. zero, if no allocations exist in, or spanning, this chunk; or
+   2. the *small* `sizeclass` of the slab in this chunk, if a `LocalAlloc` owns it; or
+   3. the *large* `sizeclass` for this chunk-spanning allocation, if no `LocalAlloc` is associated.
+
+2. A `RemoteAllocator*` indicating the `LocalAlloc` that owns the region of memory, if any.
+
+The other pointer targets either...
+
+1. the `Metaslab` record for the slab in this chunk, if it is part of a slab (either active or free in the `ChunkAllocator`), or
+2. the next free chunk in an `AddressSpaceManagerCore` per-sizeclass queue, or
+3. `nullptr` or `0` otherwise
+
+For a given address, there are five cases to consider:
+
+1. The address is not associated with `snmalloc`.  Here, the `MetaEntry`, if it is mapped, is all zeros and so...
+   * has a `sizeclass` of 0,
+   * has an all-zeros `RemoteAllocator*` (i.e., probably `nullptr`), and
+   * has an all-zeros ancillary pointer.
+
+2. The address is part of a free chunk in any of the `AddressSpaceManagerCore`s.  Here, the `MetaEntry`...
+   * has a `sizeclass` of `0`,
+   * has a `nullptr` `RemoteAllocator*`, and
+   * is carrying the pointer to the next free chunk in the same size class.
+
+3. The address is part of a live large allocation (spanning one or more 16KiB chunks).  Here, the `MetaEntry`...
+   * encodes the *large* allocation's non-zero sizeclass (and so alignment),
+   * has a `nullptr` `RemoteAllocator*`, and
+   * does not use its pointer field.
```

Comment:
It actually points to a metaslab, this is used if we free the chunk to store it in an appropriate place.  The data it contains is uninteresting if the large allocation is live, only if it is free the it points to the SlabRecord (which is reusing the space of a metaslab).